brainiak/utils/fmrisim.py

```diff
@@ -771,7 +781,19 @@ def convolve_hrf(stimfunction,
                  scale_function=True,
                  temporal_resolution=1000.0,
                  ):
-    """ Convolve the specified hrf with the timecourse
+    """ Convolve the specified hrf with the timecourse. In order
+    to accelerate the (slow) convolution, stimfunction is downsampled (mean
+    filtering) by the specified temporal resolution (e.g. a stimfunction of
+    100000 elements becomes 100 elements if the temporal_resolution is
+    1000). If temporal_resolution is 1 then the output will be the same
+    length as stimfunction.
+    Be aware that if scaling is on and events are very short
+    (> temporal_resolution * tr_duration) then the hrf may or may not come
```

Comment:
What does it mean to be ` (> temporal_resolution * tr_duration)` and "very short"? It seems that for tr_duration=2 and temporal_resolution=1000, this means something is larger than 2000. I suppose this is not the event duration, is it?

Reply:
You are right, this wasn't clear. Hopefully it is more obvious now. The point is that if the events last less than a TR you may get weird results